ludwig/utils/gbm_utils.py

```diff
@@ -0,0 +1,195 @@
+from typing import Callable, Dict, Iterable, Tuple
+
+import lightgbm as lgb
+import numpy as np
+import torch
+from lightgbm.callback import CallbackEnv
+from numpy import typing as npt
+
+from ludwig.constants import NUMBER
+from ludwig.features.base_feature import BaseFeatureMixin, OutputFeature
+from ludwig.features.category_feature import CategoryOutputFeature
+from ludwig.features.feature_utils import LudwigFeatureDict
+from ludwig.models.base import BaseModel
+
+
+def iter_feature_metrics(features: LudwigFeatureDict) -> Iterable[Tuple[str, str]]:
+    """Helper for iterating feature names and metric names."""
+    for feature_name, feature in features.items():
+        for metric in feature.metric_functions:
+            yield feature_name, metric
+
+
+def get_single_output_feature(model: BaseModel) -> BaseFeatureMixin:
+    """Helper for getting the single output feature of a model."""
+    return next(iter(model.output_features.values()))
+
+
+def sigmoid(x: npt.NDArray) -> npt.NDArray:
+    """Compute sigmoid function.
```

Comment:
Re-styled. I looked at `api.py` for reference, so we should probably change the doctoring style there too.

Reply:
It looks like this file hasn't changed yet to google-style docstrings -- perhaps an old commit?

Agreed on updating `api.py`. We discussed enforcing google-style docstrings for new code, but it would be good do a pass through old code as well.

Reply:
Could you check again? The changes should be there